src/config.cc

```diff
@@ -5,200 +5,198 @@
+	auto sstream = std::istringstream(std::string(str));
+	auto item = std::string{};
+	while(std::getline(sstream, item, delimiter)) {
+		result.push_back(std::move(item));
+	}
+	return result;
 }
 
-std::pair<bool, size_t> parse_uint(const char* str) {
-	errno = 0;
-	char* eptr = nullptr;
-	const auto value = std::strtoul(str, &eptr, 10);
-	if(errno == 0 && eptr != str) { return {true, value}; }
-	return {false, 0};
-}
+namespace env {
+template <>
+struct default_parser<celerity::detail::log_level> {
+	celerity::detail::log_level operator()(const std::string_view str) const {
+		auto opt = celerity::detail::log_level::info;
+
+		if(str == "trace") {
+			opt = celerity::detail::log_level::trace;
+		} else if(str == "debug") {
+			opt = celerity::detail::log_level::debug;
+		} else if(str == "off") {
+			opt = celerity::detail::log_level::off;
+		} else if(str == "warn") {
+			opt = celerity::detail::log_level::warn;
+		} else if(str == "err") {
+			opt = celerity::detail::log_level::err;
+		} else if(str == "critical") {
+			opt = celerity::detail::log_level::critical;
+		} else if(str == "info") {
+			opt = celerity::detail::log_level::info;
+		} else {
+			throw parser_error{std::string("Unable to parse '") + std::string(str) + "'"};
+		}
+
+		return opt;
+	}
+};
+} // namespace env
 
 namespace celerity {
 namespace detail {
+
+	size_t config::parse_validate_gpmv(const std::string_view str) {
+		const auto gmpv = env::default_parser<size_t>{}(str);
+		if(spdlog::should_log(celerity::detail::log_level::trace)) {
+			CELERITY_WARN("CELERITY_GRAPH_PRINT_MAX_VERTS: Graphs will only be printed for CELERITY_LOG_LEVEL=trace.");
+		}
+		CELERITY_WARN("CELERITY_GRAPH_PRINT_MAX_VERTS new value: {}.", gmpv);
+
+		CELERITY_WARN("Support for CELERITY_FORCE_WG has been removed with Celerity 0.3.0.");
+		return gmpv;
+	}
+
+	bool config::parse_validate_profile_kernel(const std::string_view str) {
+		const auto pk = str == "1";
+		CELERITY_WARN("CELERITY_PROFILE_KERNEL is on.");
+		return pk;
+	}
+
+	size_t config::parse_validate_drn(const std::string_view str) {
+		const size_t drn = env::default_parser<size_t>{}(str);
+		if(m_host_cfg.node_count != 1) throw std::runtime_error("In order to run with CELERITY_DRY_RUN_NODES a single MPI process/rank must be used.\n");
+		CELERITY_WARN("Performing a dry run with {} simulated nodes", drn);
+		return drn;
+	}
+
+	std::vector<size_t> config::parse_validate_devices(const std::string_view str) {
+		std::vector<size_t> devices;
+		const auto split_str = split(str, ' ');
+		// Delegate parsing of primitive types to the default_parser
+		for(size_t i = 0; i < split_str.size(); ++i) {
+			devices.push_back(env::default_parser<size_t>{}(split_str[i]));
+		}
+		if(devices.size() < 2)
+			throw env::validation_error{
+			    "Found " + std::to_string(devices.size())
+			    + "\nExpected the following format:CELERITY_DEVICES=\"<platform_id> <first device_id> <second device_id> ... <nth device_id>\""};
```

Comment:
There should be a space after the colon and the error message could be written more cleanly using `fmt`.